check_pgactivity

```diff
@@ -5600,6 +5604,167 @@ sub check_oldest_idlexact {
+                THEN 'idle_xact'
+                ELSE 'active_xact'
+              END
+          END AS kind,
+          coalesce(backend_xmin, backend_xid) AS xmin
+          FROM pg_stat_activity
+          UNION ALL (
+            -- 2PC
+            SELECT '2pc' AS kind,
+            transaction AS xmin
+            FROM  pg_prepared_xacts
+            WHERE database = current_database()
+          ) UNION ALL (
+            -- replication slots
+            SELECT 'repslot' AS kind, xmin AS xmin
+            FROM  pg_replication_slots
+            WHERE coalesce(database, current_database()) = current_database()
+          ) UNION ALL (
+            -- walsenders
+            SELECT 'walsender' AS kind, backend_xmin AS xmin
+            FROM pg_stat_replication
+          )
+        ),
+        ordered AS (
+          SELECT f.kind,
+          -- xid type doesn't have range operators as the value will wraparound.
+          -- Instead, rely on age() function and row_number() window function
+          -- to get the oldest xid found.
+          row_number() OVER (
+            PARTITION BY f.kind
+            ORDER BY age(xmin) DESC NULLS LAST
+          ) rownum, age(xmin) AS age, xmin
+          FROM raw
+          RIGHT JOIN (
+            SELECT           'query'
+            UNION ALL SELECT 'idle_xact'
+            UNION ALL SELECT 'active_xact'
+            UNION ALL SELECT '2pc'
+            UNION ALL SELECT 'repslot'
+            UNION ALL SELECT 'walsender'
+          ) f(kind) ON raw.kind = f.kind
+        )
+        SELECT kind, age, xmin FROM ordered
+        WHERE rownum = 1
+        }
+    );
+
+    # warning and critical must be raw.
+    pod2usage(
+        -message => "FATAL: critical and warning thresholds only accept raw number of transactions.",
+        -exitval => 127
+    ) unless $args{'warning'}  =~ m/^([0-9.]+)$/
+        and  $args{'critical'} =~ m/^([0-9.]+)$/;
+
+    $c_limit = $args{'critical'};
+    $w_limit = $args{'warning'};
+
+    @hosts = @{ parse_hosts %args };
+
+    pod2usage(
+        -message => 'FATAL: you must give only one host with service "oldest_xmin".',
+        -exitval => 127
+    ) if @hosts != 1;
+
+    is_compat $hosts[0], 'oldest_xmin', $PG_VERSION_84 or exit 1;
+
+    @rs = @{ query_ver( $hosts[0], %queries ) };
+
+    REC_LOOP: foreach my $r (@rs) {
+        map { $_ = 'NaN' if $_ eq ''} @{$r}[1..2];
+        push @perfdata => (
+            ["$r->[0]_age", $r->[1]],
+            ["$r->[0]_xmin", $r->[2]]
+        );
+        if (defined $c_limit) {
+            if ($r->[1] ne 'NaN' and $r->[1] > $c_limit) {
+                push @msg_crit => "$r->[0]_age";
+                next REC_LOOP;
+            }
+
+            push @msg_warn => "$r->[0]_age"
+                if ($r->[1] ne 'NaN' and $r->[1] > $w_limit);
+        }
+    }
+
+    return status_critical( $me, [
+        'Critical: '. join(',', @msg_crit)
+        . (scalar @msg_warn? 'Warning: '. join(',', @msg_warn):'')
```

Comment:
lack a space betwwen the critical list and the `Warning` label?

Reply:
Either I'm missing something or the problem exists for every other calls to status_critical.

Reply:
Most of status_critical call I found are using multiple array entries whereas you are doing a string concatenation.

Reply:
I see.  Somehow I managed to copy/paste the only broken call: https://github.com/OPMDG/check_pgactivity/blob/master/check_pgactivity#L5170-L5174

Reply:
Oops.

Fixed: a09b8b823bd0529

Thanks for the report :)